implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -80,17 +142,30 @@ private static String getValue(final String name, final Map<String, String> prop
      * instantiated in the heap.
      */
     private static Map<String, String> getEnvProperties() {
-        return unmodifiableMap(doPrivileged((PrivilegedAction<Map<String, String>>) () -> new HashMap<>(System.getenv())));
+        return doPrivileged((PrivilegedAction<Map<String, String>>) () -> new HashMap<>(System.getenv()));
     }
 
     private static int getEnvOrdinal(final Map<String, String> properties, final int ordinal) {
-        final String value = getValue(CONFIG_ORDINAL_KEY, properties);
+        String value = getValue(CONFIG_ORDINAL_KEY, properties, emptySet());
+        if (value == null) {
+            value = getValue(CONFIG_ORDINAL_KEY.toUpperCase(), properties, emptySet());
+        }
         if (value != null) {
             return Converters.INTEGER_CONVERTER.convert(value);
         }
         return ordinal;
     }
 
+    private static boolean isDottedFormat(final String propertyName) {
+        for (int i = 0; i < propertyName.length(); i++) {
+            char c = propertyName.charAt(i);
+            if (!('a' <= c && c <= 'z') && !('0' <= c && c <= '9') && c != '.') {
```

Comment:
How many times it happen to be false and how? In such case you can think to move the conditions or add new ones which can easily peel-off the common case eg  if are mostly uppercase, check if belong to that range first or anything less than `.`...

Reply:
The most usual case is looking for a dotted format property as `x.y.z`. This will let us know if we need to perform the env var conversion since we keep a cache of candidate dotted properties converted from env.